### PR TITLE
Introduce an interface to list tags

### LINKED ISSUE #243: HasUniqueTag check needs to migrate away from Skopeo
## Bug Description
This is the last check that does not use Crane directly. It will either need access to Crane (to call ListTags) or have the tags passed in via the ImageReference.

### PR DESCRIPTION
**Pull request by bcrochet**
The implementation is backed by Crane, but is used via an interface that
abstracts Crane, so that the implementation could be something else, should
that be necessary, without having to change the check.

Fixes #243

Signed-off-by: Brad P. Crochet <brad@redhat.com>

### CODE REVIEW

**Review by acornett21 — APPROVED** [review]

File: certification/internal/service/service.go [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package service
+
+type TagLister interface {
```
Comment (samira-barouti) [inline certification/internal/service/service.go:3]: IMO, using a generic name might be enable us to extend this interface for the future. I cannot think of a name that I like enough, but sth along the line of RepositoryService which includes all repo-related operations might be ok.

File: certification/internal/policy/container/has_unique_tag.go [hunk 2]
```diff
@@ -4,44 +4,33 @@ import (
...
-		log.Error("unable to execute skopeo on the image: ", err)
-		return nil, err
-	}
-
-	return runReport.Tags, nil
+	return p.TagLister.ListTags(image)
 }
 
-func (p *HasUniqueTagCheck) validate(tags []string) (bool, error) {
+func (p *HasUniqueTagCheck) validate(image string, tags []string) (bool, error) {
```
Comment (samira-barouti) [inline certification/internal/policy/container/has_unique_tag.go:33]: We are not using `image`; any reason we added this var?

File: certification/internal/service/service.go (hunk 1 above)
Comment (bcrochet) [reply certification/internal/service/service.go:3]: This is idiomatic golang. Small interfaces that are generic. This IMO is a generic name. See all the io interfaces, like Reader, Writer, Closer. They contain one method. But they might all be implemented together.

File: certification/internal/policy/container/has_unique_tag.go (hunk 2 above)
Comment (bcrochet) [reply certification/internal/policy/container/has_unique_tag.go:33]: That may have come over from the old implementation. Good catch. I'll remove it.

**Review by samira-barouti — APPROVED** [review]
/lgtm

**Review by acornett21 — APPROVED** [review]

**Pull request merged**